src/SimpleWMSBody.hpp

```diff
@@ -0,0 +1,128 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////
+//                               This file is part of CosmoScout VR                               //
+//      and may be used under the terms of the MIT license. See the LICENSE file for details.     //
+//                        Copyright: (c) 2019 German Aerospace Center (DLR)                       //
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#ifndef CSP_WMS_SIMPLE_BODIES_HPP
+#define CSP_WMS_SIMPLE_BODIES_HPP
+
+#include "WebMapTextureLoader.hpp"
+#include "utils.hpp"
+
+#include "../../../src/cs-core/SolarSystem.hpp"
+#include "../../../src/cs-core/TimeControl.hpp"
+#include "../../../src/cs-scene/CelestialBody.hpp"
+
+#include <VistaKernel/GraphicsManager/VistaOpenGLDraw.h>
+#include <VistaOGLExt/VistaBufferObject.h>
+#include <VistaOGLExt/VistaGLSLShader.h>
+#include <VistaOGLExt/VistaTexture.h>
+#include <VistaOGLExt/VistaVertexArrayObject.h>
+
+#include <glm/gtc/type_ptr.hpp>
+
+namespace csp::simpleWmsBodies {
+
+struct Properties {
+  cs::utils::Property<bool> mEnableInterpolation = true;
+  cs::utils::Property<bool> mEnableTimespan      = false;
+};
+
+/// This is just a sphere with a texture, attached to the given SPICE frame. The texture should be
+/// in equirectangular projection.
+class SimpleWMSBody : public cs::scene::CelestialBody, public IVistaOpenGLDraw {
+ public:
+  SimpleWMSBody(std::shared_ptr<cs::core::GraphicsEngine> const& graphicsEngine,
+      std::shared_ptr<cs::core::SolarSystem> const& solarSystem, std::string const& sCenterName,
+      std::string sTexture, std::string const& sFrameName, double tStartExistence,
+      double tEndExistence, std::vector<Wms> tWms,
+      std::shared_ptr<cs::core::TimeControl> timeControl,
+      std::shared_ptr<Properties>            properties = nullptr);
+  ~SimpleWMSBody() override;
+
+  /// The sun object is used for lighting computation.
+  void setSun(std::shared_ptr<const cs::scene::CelestialObject> const& sun);
+
+  /// Interface implementation of the IntersectableObject, which is a base class of
+  /// CelestialBody.
+  bool getIntersection(
+      glm::dvec3 const& rayOrigin, glm::dvec3 const& rayDir, glm::dvec3& pos) const override;
+
+  /// Interface implementation of CelestialBody.
+  double     getHeight(glm::dvec2 lngLat) const override;
+  glm::dvec3 getRadii() const override;
+
+  /// Interface implementation of IVistaOpenGLDraw.
+  bool Do() override;
+  bool GetBoundingBox(VistaBoundingBox& bb) override;
+
+  std::vector<Wms> getWms();
+  Wms              getActiveWms();
+
+  void setActiveWms(Wms wms);
+  void setActiveWms(std::string wms);
```

Comment:
I think this data can be passed by reference.

```suggestion
  std::vector<Wms> const& getWms();
  Wms              const& getActiveWms();

  void setActiveWms(Wms const& wms);
  void setActiveWms(std::string const& wms);
```